lib/unparser/emitter.rb

```diff
@@ -496,5 +520,41 @@ def run(emitter, node = node())
       emitter.new(node, self).write_to_buffer
     end
 
+    # For binary or unary operators, check whether parentheses are needed
+    # around a nested operator expression.
+    #
+    # @return [Boolean]
+    #
+    # @api private
+    #
+    def needs_parentheses?(child, emitter, side)
+      # TODO: Later this can also be used for things like if/unless/until/while
+      # statement modifiers
+      # Right now it is just for nested operator expressions
+      if emitter.operator? && self.operator?
```

Comment:
there is no need for explicit self receiver to call `#operator?` predicate. Lets kill it.